cfpb/cfgov-refresh
# Add json methods to Category model and fix migration script

To implement in-browser faceting for category pages, we need to add json
methods to the Category model to deliver some small json mapping files
for slicing answers by audience and subcategory.

We also need to update the migration script to allow the Answer object
to do the work of adding feedback modules to an Answer page, so that
when pages are created in the future (i.e. not via migration), they will
have feedback modules.

For background, see GHE ConsumerExperience/information-architecture issue 271

## Testing
If you want to see the results locally:
- Set up Ask if necessary
```
export DEPLOY_ENVIRONMENT=build
./drop-db.sh
./create-mysql-db.sh
mysql -uroot v1 < ~/Downloads/production_django.sql
python cfgov/manage.py migrate
python cfgov/manage.py runscript migrate_knowledgebase
```

- Visit a category page, such as http://localhost:8000/ask-cfpb/category-credit-cards/, and your browser's dev tools console should allow you to inspect three json mapping files: 
- `audienceMap`
-  `subcatMap`
-  `answerMap`  

These objects are global for now, but will probably change name and scope after implementation.

## Migration sequencing
I'll need to regenerate the migration in this PR after PR #2857 is merged.
